### PR TITLE
TSConverter refactoring

### PR DESCRIPTION
**Pull request by ZikkeyLS**
Some test refactoring, can include broken code :)
Тут нет важных изменений. Нашёл ещё много мест, которые можно как-то переписать, но не хватает знаний c++, чтобы сделать лучше.

### CODE REVIEW

File: skyrim-platform/src/platform_se/codegen/TSConverter.cpp [hunk 1]
```diff
@@ -86,48 +86,59 @@ int main(int argc, char* argv[])
 
   auto parseReturnValue = [&](std::string rawType,
                               std::string objectTypeName) -> std::string {
-    if (rawType == "Int" || rawType == "Float") {
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/codegen/TSConverter.cpp:89]: Please could you bring back the version with `if`. This one seems incorrect

File: skyrim-platform/src/platform_se/codegen/TSConverter.cpp [hunk 2]
```diff
@@ -182,43 +193,45 @@ int main(int argc, char* argv[])
 
   std::function<void(nlohmann::json)> dumpType =
     [&](nlohmann::json data) -> void {
-    if (ignored.contains(data.at("name").get<std::string>()) ||
-        dumped.contains(data.at("name").get<std::string>())) {
+
+    std::string name = data.at("name").get<std::string>();
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/codegen/TSConverter.cpp:197]: `name` and `dataName` has the same meaning

File: skyrim-platform/src/platform_se/codegen/TSConverter.cpp (hunk 1 above)
Comment (ZikkeyLS) [reply skyrim-platform/src/platform_se/codegen/TSConverter.cpp:89]: ready

File: skyrim-platform/src/platform_se/codegen/TSConverter.cpp (hunk 2 above)
Comment (ZikkeyLS) [reply skyrim-platform/src/platform_se/codegen/TSConverter.cpp:187]: ready

File: skyrim-platform/src/platform_se/codegen/TSConverter.cpp [hunk 3]
```diff
@@ -86,6 +86,7 @@ int main(int argc, char* argv[])
 
   auto parseReturnValue = [&](std::string rawType,
                               std::string objectTypeName) -> std::string {
+
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/codegen/TSConverter.cpp:89]: Could you please remove this whitespace?

Comment (Pospelove) [reply skyrim-platform/src/platform_se/codegen/TSConverter.cpp:89]: I think it's going to be merged after that. LGTM.

**Comment by Milord-Freeman** [comment]
>  Нашёл ещё много мест
You can mark these places and describe the essence of the problem, putting it up for discussion. 
Maybe in another PR.

File: skyrim-platform/src/platform_se/codegen/TSConverter.cpp (hunk 3 above)
Comment (ZikkeyLS) [reply skyrim-platform/src/platform_se/codegen/TSConverter.cpp]: Ready

**Review by Pospelove — APPROVED** [review]
LGTM

**Comment by Pospelove** [comment]
@Milord-Freeman Could you please confirm that PR doesn't break the build system on your machine?

**Comment by Milord-Freeman** [comment]
изображение
All project doesn't buildable, TSConverter seems fine, but codegen target failed. Please test your solution on real data before commit changes.

**Comment by Pospelove** [comment]
Closing for now. Feel free to reopen when you would be able to continue

**Pull request closed without merging**